### PR TITLE
Prepare support for PySide6 drawLines and friends

### PR DESCRIPTION
**Pull request by pijyoi**
As PYSIDE-1924 (https://bugreports.qt.io/browse/PYSIDE-1924) feature has been added, we can prepare pyqtgraph to make use of it when it gets into a released version.
For now, testing can be done with the wheels at https://download.qt.io/snapshots/ci/pyside/dev/latest/

One issue is that PySide6's array API implementation will requirement passing of different arguments than the other codepaths. For this, we add a `drawargs()` method.

A synthetic benchmark to exercise the new API.
```python
import numpy as np
import pyqtgraph as pg
from pyqtgraph.Qt import QtCore, QtGui

from time import perf_counter

def generate_pattern(nsegs, size):
    rng = np.random.default_rng()
    x = rng.random(nsegs) * size
    y = rng.random(nsegs) * size
    arr = np.empty((nsegs, 4), dtype=np.float64)
    arr[:, 0] = x
    arr[:, 1] = y
    arr[:, 2] = x + 2
    arr[:, 3] = y + 2
    return arr

nsegs = 10_000
size = 500
nframes = 100
pattern = generate_pattern(nsegs, size)

def run(use_array):
    # generate lines once
    segments = pg.Qt.internals.PrimitiveArray(QtCore.QLineF, 4, use_array=use_array)
    segments.resize(nsegs)
    memory = segments.ndarray()
    memory[:] = pattern

    # draw multiple frames using the same lines array
    t0 = perf_counter()
    for _ in range(nframes):
        qimg = QtGui.QImage(size, size, QtGui.QImage.Format.Format_RGB32)
        qimg.fill(QtCore.Qt.GlobalColor.transparent)
        painter = QtGui.QPainter(qimg)
        painter.setPen(QtCore.Qt.GlobalColor.cyan)
        painter.drawLines(*segments.drawargs())
        painter.end()
    t1 = perf_counter()
    return t1 - t0

for use_array in [False, None, True]:
    duration = run(use_array)
    fps = int(nframes / duration)
    print(f'{use_array=} {duration=:.3f} {fps=}')
```

### CODE REVIEW

**Comment by pijyoi** [comment]
Added re-use of backing memory if new size is smaller than largest previous sizes.
This falls back to re-allocation when using PyQt sip.array as sip.array(s) will eventually segfault when sliced.

**Comment by pijyoi** [comment]
The re-use of backing memory now also works with the unreleased `pyqt5-sip >= 12.11.2` and `pyqt6-sip >= 13.4.2`. 
https://riverbankcomputing.com/pypi/simple/
It is not clear that those versions will be released at all as newer versions already exist (12.12.0 and 13.5.0).

To exercise this PR more within the CI, a temporary change was made to set `'segmentedLineMode': 'on'`.
Obviously, that should not be merged.

**Comment by j9ac9k** [comment]
@pijyoi i love this diff, thank you for commenting the `internals.py` so thoroughly.

Since we just released 0.13.2, I would be good w/ merging this PR (once `segmentedLineMode` is set back to `auto`).

File: pyqtgraph/__init__.py [hunk 1]
```diff
@@ -55,7 +55,7 @@
                                  # change in the future.
     'useCupy': False,  # When True, attempt to use cupy ( currently only with ImageItem and related functions )
     'useNumba': False, # When True, use numba
-    'segmentedLineMode': 'auto',  # segmented line mode, controls if lines are plotted in segments or continuous
```
Comment (j9ac9k) [inline pyqtgraph/__init__.py:58]: Does this need to be reverted before we can merge?

Comment (pijyoi) [reply pyqtgraph/__init__.py:58]: Yes, it does. But as the release of PySide6 6.4.3 is imminent, I thought I would just wait to test out against the official release and fix up the comments regarding working version numbers.

Comment (j9ac9k) [reply pyqtgraph/__init__.py:58]: Oh right, keep forgetting 6.4.3 hasn't been released yet!

Comment (pijyoi) [reply pyqtgraph/__init__.py]: The 6.4.3 wheels are now available at https://download.qt.io/snapshots/ci/pyside/6.4/latest/split_wheels/ but not yet on pypi.org

**Comment by j9ac9k** [comment]
PySide 6.5.0 has been released, going to close/re-open this PR to trigger a new CI run.

**Comment by j9ac9k** [comment]
unrelated to this PR, but looks like we're getting a failure in CI

```
Run xvfb-run --server-args="-screen 0, 1920x1200x24 -ac +extension GLX +render -noreset" python -m pyqtgraph.util.glinfo
[14](https://github.com/pyqtgraph/pyqtgraph/actions/runs/4610114611/jobs/8148181442?pr=2596#step:10:15)
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
[15](https://github.com/pyqtgraph/pyqtgraph/actions/runs/4610114611/jobs/8148181442?pr=2596#step:10:16)
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
[16](https://github.com/pyqtgraph/pyqtgraph/actions/runs/4610114611/jobs/8148181442?pr=2596#step:10:17)

[17](https://github.com/pyqtgraph/pyqtgraph/actions/runs/4610114611/jobs/8148181442?pr=2596#step:10:18)
Available platform plugins are: xcb, eglfs, offscreen, vnc, wayland, minimalegl, linuxfb, vkkhrdisplay, minimal, wayland-egl.
[18](https://github.com/pyqtgraph/pyqtgraph/actions/runs/4610114611/jobs/8148181442?pr=2596#step:10:19)

[19](https://github.com/pyqtgraph/pyqtgraph/actions/runs/4610114611/jobs/8148181442?pr=2596#step:10:20)
Aborted (core dumped)
```

**Comment by j9ac9k** [comment]
Closing and Re-Opening again now that Qt 6.5 works in CI.

**Comment by j9ac9k** [comment]
Looks like CI for pyside 6.5 is green across the board, let's undo the segmentation line check and merge 🎊

**Comment by pijyoi** [comment]
It's already undone

**Comment by j9ac9k** [comment]
> It's already undone

oh... well, that's silly of me.  Thanks for your work on this @pijyoi !

**Pull request merged**